### ISSUE TITLE
[Process exited 0]

### ISSUE THREAD

**Issue by sassanh** [airodactyl/neovim-ranger#2]
Whenever I hit <kbd>cr</kbd> on a file to open it, before seeing the file opened in nvim, I see an empty buffer with just a short message in it:

```
[Process exited 0]
```

I know it's because nvim's terminal but it'd be great if we could get over it.

**Comment by Enzime** [comment]
Yeah, I looked into fixing this, and I don't think it's possible right now with neovim's TUI, but I think they're planning to allow skipping this, so we'll just have to wait for them to implement this.

**Comment by sassanh** [comment]
I found this which seems to solve this problem, do you understand how: https://github.com/panickbr/neovim-ranger/blob/master/autoload/rangerchooser.vim
I think it's closing the buffer after choosing the file.

**Comment by Enzime** [comment]
I'll take a look at this later, and see if I can incorporate anything from there. I'm curious, are you using my repository, or the one you linked?

**Comment by sassanh** [comment]
I was using yours but as the mentioned error was kinda annoying now I'm using another one https://github.com/francoiscabrol/ranger.vim

**Comment by vqng** [comment]
any solution yet? this is very annoying bug, I can't switch buffer using `:b#` now.

**Comment by matthewgrossman** [comment]
`autocmd TermClose term://.//*:ranger* bprevious | bwipeout! #` is what I've been using, and it mostly works

**Comment by Tej12345** [comment]
```
 Plug 'francoiscabrol/ranger.vim'
 let g:ranger_map_keys = 0
 nnoremap <leader>1 :call RangerOpen()<CR>
 function! RangerOpen() 
     let g:ranger_launched_buffer = bufname("%")
     execute "Ranger"
 endfunction
 autocmd TermClose term://.//*:ranger* call RangerClose()
 function! RangerClose()
     execute 'buffer' g:ranger_launched_buffer | bd! #
     "execute 'buffer' g:ranger_launched_buffer | execute "bw!" bufname("ranger")
 endfunction
```
Thank me later

"#" - means previous buffer
"%" - means current buffer

**Comment by QiangZiBro** [comment]
> `autocmd TermClose term://.//*:ranger* bprevious | bwipeout! #` is what I've been using, and it mostly works

This works for me
